x-pack/plugins/data_visualizer/server/routes/routes.ts

```diff
@@ -0,0 +1,265 @@
+  samplerShardSize: number,
+  timeFieldName: string | undefined,
+  earliestMs: number | undefined,
+  latestMs: number | undefined,
+  runtimeMappings: estypes.RuntimeFields
+) {
+  const dv = new DataVisualizer(client);
+  return dv.getOverallStats(
+    indexPatternTitle,
+    query,
+    aggregatableFields,
+    nonAggregatableFields,
+    samplerShardSize,
+    timeFieldName,
+    earliestMs,
+    latestMs,
+    runtimeMappings
+  );
+}
+
+function getStatsForFields(
+  client: IScopedClusterClient,
+  indexPatternTitle: string,
+  query: any,
+  fields: Field[],
+  samplerShardSize: number,
+  timeFieldName: string | undefined,
+  earliestMs: number | undefined,
+  latestMs: number | undefined,
+  interval: number | undefined,
+  maxExamples: number,
+  runtimeMappings: estypes.RuntimeFields
+) {
+  const dv = new DataVisualizer(client);
+  return dv.getStatsForFields(
+    indexPatternTitle,
+    query,
+    fields,
+    samplerShardSize,
+    timeFieldName,
+    earliestMs,
+    latestMs,
+    interval,
+    maxExamples,
+    runtimeMappings
+  );
+}
+
+function getHistogramsForFields(
+  client: IScopedClusterClient,
+  indexPatternTitle: string,
+  query: any,
+  fields: HistogramField[],
+  samplerShardSize: number,
+  runtimeMappings: estypes.RuntimeFields
+) {
+  const dv = new DataVisualizer(client);
+  return dv.getHistogramsForFields(
+    indexPatternTitle,
+    query,
+    fields,
+    samplerShardSize,
+    runtimeMappings
+  );
+}
+/**
+ * Routes for the index data visualizer.
+ */
+export function dataVisualizerRoutes(coreSetup: CoreSetup<StartDeps, unknown>, logger: Logger) {
+  const router = coreSetup.http.createRouter();
+
+  /**
+   * @apiGroup DataVisualizer
+   *
+   * @api {post} /internal/data_visualizer/get_field_histograms/:indexPatternTitle Get histograms for fields
+   * @apiName GetHistogramsForFields
+   * @apiDescription Returns the histograms on a list fields in the specified index pattern.
+   *
+   * @apiSchema (params) indexPatternTitleSchema
+   * @apiSchema (body) dataVisualizerFieldHistogramsSchema
+   *
+   * @apiSuccess {Object} fieldName histograms by field, keyed on the name of the field.
+   */
+  router.post(
+    {
+      path: '/internal/data_visualizer/get_field_histograms/{indexPatternTitle}',
+      validate: {
+        params: indexPatternTitleSchema,
+        body: dataVisualizerFieldHistogramsSchema,
+      },
+    },
+    async (context, request, response) => {
+      try {
+        const {
+          params: { indexPatternTitle },
+          body: { query, fields, samplerShardSize, runtimeMappings },
+        } = request;
+
+        const results = await getHistogramsForFields(
+          context.core.elasticsearch.client,
+          indexPatternTitle,
+          query,
+          fields,
+          samplerShardSize,
+          runtimeMappings
+        );
+
+        return response.ok({
+          body: results,
+        });
+      } catch (e) {
+        logger.warn(e);
```

Comment:
This should probably be `logger.error`
but i'm not sure we should be just throwing the error to the log like this without any context of where it has come from.
As these errors are passed to the client for display, do we even need to add them to the kibana log?

Reply:
Removed logging and Logger here https://github.com/elastic/kibana/pull/100922/commits/403a8e38d5d2af99f3081e68294c9ad35dfe270b